src/identity.ts

```diff
@@ -0,0 +1,227 @@
+        )}" resolves to more than one identity. Either use "getIdentities" to list all of the matching identities, or remove redundant identity using "everynode identity rm" or "deleteIdentity"`
+      );
+    }
+
+    if (identities.items.length == 0) {
+      throw new Error('No credentials found');
+    }
+
+    debug(`${JSON.stringify(identityOrIdsOrTags)}: returning ${identities.items[0].id}`);
+
+    identityId = identities.items[0].id;
+  }
+
+  const creds = await getTokenForIdentity(serviceId, identityId);
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any,security/detect-object-injection
+  return (provider as any)[serviceId].normalize(creds);
+};
+
+/**
+ * Search for matching identities and return all matches found. The results do not include security tokens.
+ *
+ * @param serviceId The service to search for matching identities within.
+ * @param idsOrTags Either a { userId, tenantId } that can be used to search for a matching identity, or a set of
+ * tags that will be used to search.
+ * @param options Options to control the pagination or request subsequent pages of results.
+ * @return A set of matching identities
+ */
+export const getIdentities = async (
+  serviceId: string,
+  idsOrTags: IEveryAuthUserTenantSet | IEveryAuthTagSet,
+  options?: IEveryAuthSearchOptions
+): Promise<IEveryAuthIdentitySearch> => {
+  let identities;
+
+  if (idsOrTags.userId || idsOrTags.tenantId) {
+    identities = await getIdentitiesByTags(
+      serviceId,
+      {
+        [SERVICE_TAG]: serviceId,
+        ...('userId' in idsOrTags ? { [USER_TAG]: idsOrTags.userId } : {}),
+        ...('tenantId' in idsOrTags ? { [TENANT_TAG]: idsOrTags.tenantId } : {}),
+      },
+      options
+    );
+  } else {
+    identities = await getIdentitiesByTags(serviceId, idsOrTags as IEveryAuthTagSet, options);
+  }
+
+  debug(`${JSON.stringify(idsOrTags)}: Found ${identities.items.length} matching identities`);
+
+  // Sanitize the return set.
+  return {
+    items: identities.items.map((identity) => ({
+      id: identity.id,
+      tags: identity.tags,
+      dateModified: identity.dateModified,
+    })),
+    next: identities.next,
+  };
+};
+
+/**
+ * Delete a specified identity.
+ *
+ * @param serviceId The service to search for matching identities within.
+ * @param identityId The identity to delete.
+ */
+export const deleteIdentity = async (serviceId: string, identityId: string): Promise<void> => {
+  const profile = await getAuthedProfile();
+  const baseUrl = `${profile.baseUrl}/v2/account/${profile.account}/subscription/${profile.subscription}`;
+  const conUrl = `/connector/${serviceId}`;
+  const intUrl = '/integration/everyauth';
+
+  // Get the identity object
+  const identity = await superagent
+    .get(`${baseUrl}${conUrl}/identity/${identityId}`)
+    .set('User-Agent', EveryAuthVersion)
+    .set('Authorization', `Bearer ${profile.token}`)
+    .ok(() => true);
+
+  if (identity.statusCode != 200) {
+    return;
```

Comment:
Why not throw? How is the caller supposed to know this has failed?

Reply:
Changed to only throw on > 299 and not 404.